lib/cn/move.c

```diff
@@ -29,54 +29,62 @@ cn_move(
     bool                       src_del,
     struct cn_tree_node       *tgt_node)
 {
-    struct cn_tree *tree;
-    struct kvset_list_entry *src, *src_end, *tgt;
+    struct kvset_list_entry *src, *tgt;
     struct list_head *src_head, *tgt_head;
-    uint64_t *src_ksidv = NULL;
+    struct cn_tree *tree;
     uint64_t src_nodeid;
-    merr_t err;
 
     INVARIANT(w && src_node && tgt_node);
-    INVARIANT((src_cnt > 0 && src_list) || (src_cnt == 0 && !src_list));
+    INVARIANT(!src_cnt == !src_list);
```

Comment:
Sweet!

Reply:
I like having assert macros to handle "if and only if" and "implies" logic:
```
assert_iff(src_cnt, src_list);
assert_imply(len > 0, data != NULL);
```

Reply:
Fine, but don't wander into any dark alleys..  :-)